tests/v0_tests/test_client.py

```diff
@@ -54,14 +58,58 @@ def test_version_check_instantiation(self):
             self.assertIn(self.client_settings['url'], marqo_url_and_version_cache)
             assert marqo_url_and_version_cache[self.client_settings['url']] == '0.0.0'
 
+    def test_skip_version_check_for_cloud_v2(self):
+        for url in ["https://api.marqo.ai", "https://cloud.marqo.ai"]:
+            with mock.patch("marqo.client.mq_logger.warning") as mock_warning,\
+                mock.patch("marqo.client.Client.get_marqo") as mock_get_marqo:
+
+                mock_get_marqo.return_value = {'version': '0.0.0'}
+                client = Client(url = url)
+
+                mock_get_marqo.assert_not_called()
+
+                assert url in marqo_url_and_version_cache
+                assert marqo_url_and_version_cache[url] == '_skipped'
+
+    def test_skip_version_check_for_previously_labelled_url(self):
+        with mock.patch.dict("marqo.client.marqo_url_and_version_cache", {self.client_settings["url"]:"_skipped"}) as mock_cache,\
+            mock.patch("marqo.client.Client.get_marqo") as mock_get_marqo:
+            client = Client(**self.client_settings)
+
+            mock_get_marqo.assert_not_called()
+
+    def test_error_handling_in_version_check(self):
+        side_effect_list = [requests.exceptions.JSONDecodeError("test", "test", 1), BackendCommunicationError("test"),
+                            BackendTimeoutError("test"), requests.exceptions.RequestException("test"),
+                            KeyError("test"), KeyError("test"), requests.exceptions.Timeout("test")]
+        for i, side_effect in enumerate(side_effect_list):
+            with mock.patch("marqo.client.mq_logger.warning") as mock_warning, \
+                    mock.patch("marqo.client.Client.get_marqo") as mock_get_marqo:
+                mock_get_marqo.side_effect = side_effect
+                client = Client(**self.client_settings)
+
+                mock_get_marqo.assert_called_once()
+
+                # Check the warning was logged
+                mock_warning.assert_called_once()
+
+                # Get the warning message
+                warning_message = mock_warning.call_args[0][0]
+
+                # Assert the message is what you expect
+                self.assertIn("Marqo encountered a problem trying to check the Marqo version found", warning_message)
+                self.assertEqual(marqo_url_and_version_cache, dict({self.client_settings["url"] : "_skipped"}))
+
+                marqo_url_and_version_cache.clear()
+
     def test_version_check_multiple_instantiation(self):
```

Comment:
Can we have one test similar to this but for skipped scenario, where we make sure multiple instantiations make only one `get_marqo` attempt and also log a warning?